### PR TITLE
Cdi plot

### PR DESCRIPTION
**Pull request by licode**
Add plotting function to cdi reconstruction.

### CODE REVIEW

File: skxray/cdi.py [hunk 1]
```diff
@@ -281,6 +283,10 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
     n_iterations : int, optional
         number of iterations to run.
         default is 1000.
+    plot_function : function, optional
+        plotting function
```
Comment (ericdill) [inline skxray/cdi.py]: Note that this is a callback function that expects to receive these four objects: `sample_obj, obj_error, diff_error, sup_error`

File: skxray/cdi.py [hunk 2]
```diff
@@ -281,6 +283,10 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
...
+    plot_step : int, optional
+        step interval for plotting
```
Comment (ericdill) [inline skxray/cdi.py]: what is the step interval?

File: skxray/cdi.py [hunk 3]
```diff
@@ -360,6 +366,9 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
                     sup_error[n] = np.sum(sup_old)
                     sup_old = np.array(sup)
 
+        if plot_function and np.mod(n_iterations, plot_step) == 0:
```
Comment (ericdill) [inline skxray/cdi.py]: could also do `n_iterations % plot_step == 0` instead of `np.mod`

**Comment by licode** [comment]
Done. thanks!

File: skxray/cdi.py [hunk 4]
```diff
@@ -45,6 +45,7 @@
 import numpy as np
 import time
 from scipy.ndimage.filters import gaussian_filter
+import matplotlib.pyplot as plt
```
Comment (tacaswell) [inline skxray/cdi.py]: Do not need this import here

File: skxray/cdi.py [hunk 5]
```diff
@@ -239,7 +240,8 @@ def generate_disk_support(sup_radius, shape_v):
 def cdi_recon(diffracted_pattern, sample_obj, sup,
               beta=1.15, start_avg=0.8, pi_modulus_flag='Complex',
               sw_flag=True, sw_sigma=0.5, sw_threshold=0.1, sw_start=0.2,
-              sw_end=0.8, sw_step=10, n_iterations=1000):
+              sw_end=0.8, sw_step=10, n_iterations=1000,
+              plot_function=None, plot_step=10):
```
Comment (tacaswell) [inline skxray/cdi.py]: I would not call it `plot_function` as the function could do _anything_.

**Comment by licode** [comment]
Done!

File: skxray/cdi.py [hunk 6]
```diff
@@ -281,6 +282,12 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
     n_iterations : int, optional
         number of iterations to run.
         default is 1000.
+    cb_function : function, optional
+        This is a callback function that expects to receive these
+        four objects: sample_obj, obj_error, diff_error, sup_error
```
Comment (tacaswell) [inline skxray/cdi.py]: You should also list the expected types.  Imaging reading this doc string with out having read the function, but wanting to write a callback to plug into this.

File: skxray/cdi.py [hunk 7]
```diff
@@ -281,6 +282,12 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
...
+    plot_step : int, optional
```
Comment (tacaswell) [inline skxray/cdi.py]: change the 'plot' here -> 'cb'?

**Comment by tacaswell** [comment]
Two minor doc/naming comments, but :+1: to merge as soon as those are fixed.

**Comment by licode** [comment]
Updated.

**Pull request merged**